addons/website/static/src/js/editor/snippets.editor.js

```diff
@@ -555,7 +555,20 @@ const wSnippetMenu = weSnippetEditor.SnippetsMenu.extend({
             selectedTextEl.classList.add(...optionClassList);
             let $snippet = null;
             try {
-                range.surroundContents(selectedTextEl);
+                const backgroundColorParentEl =
+                    range.commonAncestorContainer.parentNode
+                    ?.closest('font[style*="background-color"], font[style*="background-image"], font[class^="bg-"]');
+                if (
+                    selectedTextEl.matches(".o_animated_text") && backgroundColorParentEl &&
+                    backgroundColorParentEl.innerHTML === range.commonAncestorContainer.textContent
+                ) {
+                    // To keep the background color applied, the "<font>" element must stay inside
+                    // "o_animated_text" due to the "inline-block" display.
+                    backgroundColorParentEl.replaceWith(selectedTextEl);
```

Comment:
Maybe you can avoid the security alert with that. To test...

```suggestion
                    const parent = backgroundColorParentEl.parentNode;
                    if (parent) {
                        parent.insertBefore(selectedTextEl, backgroundColorParentEl);
                        selectedTextEl.appendChild(backgroundColorParentEl);
                    }
```